packages/plugins/documentation/server/services/helpers/build-api-endpoint-path.js

```diff
@@ -120,61 +128,75 @@ const getPaths = ({ routeInfo, attributes, tag }) => {
     return acc;
   }, {});
 
-  return { paths };
+  return paths;
 };
 
 /**
- * @description - Builds the Swagger paths object for each api
+ * @decription Gets all open api paths object for a given content type
  *
- * @param {object} api - Information about the current api
- * @property {string} api.name - The name of the api
- * @property {string} api.getter - The getter for the api (api | plugin)
- * @property {array} api.ctNames - The name of all contentTypes found on the api
+ * @param {object} apiInfo
+ * @property {string} apiInfo.name - The name of the api
+ * @property {string} apiInfo.getter - api | plugin
+ * @property {array} apiInfo.ctNames - All contentType names on the api
+ * @property {string} apiInfo.uniqueName - Api name | Api name + Content type name
+ * @property {object} apiInfo.attributes - Attributes on content type
+ * @property {object} apiInfo.routeInfo - The routes for the api
  *
- * @returns {object}
+ * @returns {object} Open API paths
  */
-module.exports = api => {
-  if (!api.ctNames.length && api.getter === 'plugin') {
-    // Set arbitrary attributes
-    const attributes = { foo: { type: 'string' } };
-    const routeInfo = strapi.plugin(api.name).routes['admin'];
-
-    const apiInfo = {
-      routeInfo,
-      attributes,
-      tag: api.name,
-    };
-
-    return getPaths(apiInfo);
-  }
-
-  // An api could have multiple contentTypes
+const getAllPathsForContentType = ({
+  name,
+  getter,
+  ctNames,
+  uniqueName,
+  routeInfo,
+  attributes,
+}) => {
   let paths = {};
-  for (const contentTypeName of api.ctNames) {
-    // Get the attributes found on the api's contentType
-    const uid = `${api.getter}::${api.name}.${contentTypeName}`;
-    const ct = strapi.contentType(uid);
-    const attributes = ct.attributes;
-
-    // Get the routes for the current api
-    const routeInfo =
-      api.getter === 'plugin'
-        ? strapi.plugin(api.name).routes['content-api']
-        : strapi.api[api.name].routes[contentTypeName];
-
-    // Parse an identifier for OpenAPI tag if the api name and contentType name don't match
-    const tag = api.name === contentTypeName ? api.name : `${api.name} - ${contentTypeName}`;
+  if (!ctNames.length && getter === 'plugin') {
+    const routeInfo = strapi.plugin(name).routes['admin'];
```

Comment:
The plugin can only generate documentation based on the attributes in a schema.

The original check was to generate arbitrary attributes for a plugin that doesn't have any content types. Overrides would allow the user to or plugin developer to specify them by hand. Unfortunately overrides don't work right now.

I think actually we can remove these checks and just not document anything that doesn't have a schema.json / attributes. Having documentation for an endpoint with just `foo: string` isn't very useful.